pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -715,54 +772,59 @@ class CreateVmModal extends React.Component {
                     providerName={providerName}
                     source={this.state.source}
                     sourceType={this.state.sourceType}
+                    os={this.state.os}
+                    osInfoList={this.props.osInfoList}
+                    downloadOSSupported={this.isVirtInstallGreaterThan220()}
                     onValueChanged={this.onValueChanged}
                     validationFailed={validationFailed} />
 
                 <hr />
 
-                { this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
+                {this.state.sourceType != DOWNLOAD_AN_OS &&
                 <>
-                    <StorageRow
-                        connectionName={this.state.connectionName}
-                        storageSize={this.state.storageSize}
-                        storageSizeUnit={this.state.storageSizeUnit}
+                    <OSRow
```

Comment:
Why is this component called from 2 different places? From here and from SourceRow?